test/testgpu_with_sdlrender.c

```diff
@@ -0,0 +1,203 @@
+
+    renderer = SDL_CreateRendererWithProperties(
+        props);
+
+    if (!renderer) {
+        SDL_Log("SDL_CreateRendererWithProperties failed: %s", SDL_GetError());
+        return SDL_APP_FAILURE;
+    }
+
+    SDL_Surface *bmp_surf = SDL_LoadBMP("icon.bmp");
+
+    if (!bmp_surf) {
+        SDL_Log("SDL_LoadBMP failed: %s", SDL_GetError());
+        return SDL_APP_FAILURE;
+    }
+
+    /* Creation of any resources with manually controlled GPU renderer require a command buffer to be set*/
+    SDL_GPUCommandBuffer *cmdbuf = SDL_AcquireGPUCommandBuffer(gpu_device);
+
+    if (cmdbuf == NULL) {
+        SDL_Log("SDL_AcquireGPUCommandBuffer failed: %s", SDL_GetError());
+        return SDL_APP_FAILURE;
+    }
+
+    if (!SDL_SetRenderGPUCommandBuffer(renderer, cmdbuf)) {
+        SDL_Log("SDL_SetRenderGPUCommandBuffer failed: %s", SDL_GetError());
+        return SDL_APP_FAILURE;
+    }
+
+    icon_texture = SDL_CreateTextureFromSurface(renderer, bmp_surf);
+
+    SDL_SubmitGPUCommandBuffer(cmdbuf);
+
+    SDL_DestroySurface(bmp_surf);
+
+    if (!icon_texture) {
+        SDL_Log("SDL_LoadBMP failed: %s", SDL_GetError());
+        return SDL_APP_FAILURE;
+    }
+
+    then = SDL_GetTicks();
+
+    return SDL_APP_CONTINUE;
+}
+
+SDL_AppResult SDL_AppEvent(void *appstate, SDL_Event *event)
+{
+    if (event->type == SDL_EVENT_KEY_DOWN) {
+        icon_pos_x += 16.0f;
+    }
+
+    return SDLTest_CommonEventMainCallbacks(state, event);
+}
+
+SDL_AppResult SDL_AppIterate(void *appstate)
+{
+    SDL_GPUCommandBuffer *cmdbuf = SDL_AcquireGPUCommandBuffer(gpu_device);
+
+    if (cmdbuf == NULL) {
+        SDL_Log("SDL_AcquireGPUCommandBuffer failed: %s", SDL_GetError());
+        return SDL_APP_FAILURE;
+    }
+
+    SDL_GPUTexture *swapchainTexture;
+    if (!SDL_WaitAndAcquireGPUSwapchainTexture(cmdbuf, state->windows[0], &swapchainTexture, NULL, NULL)) {
+        SDL_Log("SDL_AcquireGPUSwapchainTexture failed: %s", SDL_GetError());
+        return SDL_APP_FAILURE;
+    }
+
+    if (swapchainTexture != NULL) {
+        const double currentTime = (double)SDL_GetPerformanceCounter() / SDL_GetPerformanceFrequency();
+        SDL_GPURenderPass *renderPass;
+        SDL_GPUColorTargetInfo color_target_info;
+        SDL_zero(color_target_info);
+        color_target_info.texture = swapchainTexture;
+        color_target_info.clear_color.r = (float)(0.5 + 0.5 * SDL_sin(currentTime));
+        color_target_info.clear_color.g = (float)(0.5 + 0.5 * SDL_sin(currentTime + SDL_PI_D * 2 / 3));
+        color_target_info.clear_color.b = (float)(0.5 + 0.5 * SDL_sin(currentTime + SDL_PI_D * 4 / 3));
+        color_target_info.clear_color.a = 1.0f;
+        color_target_info.load_op = SDL_GPU_LOADOP_CLEAR;
+        color_target_info.store_op = SDL_GPU_STOREOP_STORE;
+
+        /* Custom GPU rendering */
+        renderPass = SDL_BeginGPURenderPass(cmdbuf, &color_target_info, 1, NULL);
+        /* Render Half-Life 3 or whatever */
```

Comment:
I may need to put at least a basic triangle render code here to verify that it actually works, but I presume if swapchain clear color is preserved after SDL_RenderPresentToGPUTexture - we are fine to go?